File: refinery-service-java/src/main/java/hu/zoltanterek/worldsim/refinery/model/PatchResponse.java
package hu.zoltanterek.worldsim.refinery.model;

import java.util.List;

public record PatchResponse(
        String schemaVersion,
        String requestId,
        long seed,
        List<PatchOp> patch,
        List<String> explain,
        List<String> warnings
) {
}

File: WorldSim.Contracts/v1/PatchResponse.cs
using System.Text.Json.Serialization;

namespace WorldSim.Contracts.V1;

public sealed record PatchResponse(
    [property: JsonPropertyName("schemaVersion")] string SchemaVersion,
    [property: JsonPropertyName("requestId")] string RequestId,
    [property: JsonPropertyName("seed")] long Seed,
    [property: JsonPropertyName("patch")] IReadOnlyList<PatchOp> Patch,
    [property: JsonPropertyName("explain")] IReadOnlyList<string> Explain,
    [property: JsonPropertyName("warnings")] IReadOnlyList<string> Warnings
);
